pkg/build/api/v1/types.go

```diff
@@ -115,6 +117,12 @@ type BuildSource struct {
 	// Type of source control management system
 	Type BuildSourceType `json:"type" description:"type of source control management system"`
 
+	// Dockerfile is the raw contents of a Dockerfile which should be built. When this option is
+	// specified, the From and Env on the Docker build strategy are applied on top of this file.
+	// The Dockerfile source type may be used with other options like .git - in those cases the
+	// Git repo will have any innate Dockerfile replaced in the context dir.
+	Dockerfile *string `json:"dockerfile,omitempty" description:"the contents of a Dockerfile to build; FROM and ENV may be overriden if you have specified 'from' and 'env' on the build strategy"`
```

Comment:
That darn d

On Sep 17, 2015, at 11:59 PM, Jordan Liggitt <notifications@github.com>
wrote:

In pkg/build/api/v1/types.go
<https://github.com/openshift/origin/pull/4717#discussion_r39822398>:

> @@ -115,6 +117,12 @@ type BuildSource struct {
>  	// Type of source control management system
>  	Type BuildSourceType `json:"type" description:"type of source control management system"`
>
> +	// Dockerfile is the raw contents of a Dockerfile which should be built. When this option is
> +	// specified, the From and Env on the Docker build strategy are applied on top of this file.
> +	// The Dockerfile source type may be used with other options like .git - in those cases the
> +	// Git repo will have any innate Dockerfile replaced in the context dir.
> +	Dockerfile *string `json:"dockerfile,omitempty" description:"the contents of a Dockerfile to build; FROM and ENV may be overriden if you have specified 'from' and 'env' on the build strategy"`

overridden

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4717/files#r39822398>.